packages/react-components/src/utils/setupConfiguration.ts

```diff
@@ -193,16 +218,16 @@ async function fetchPassword(passwordKey: string): Promise<string> {
             return password._properties.clear_password;
         }
     }
-    return '';
+    return undefined;
 }
 
 async function fetchApiKey(): Promise<string> {
-    return fetchPassword(PasswordKeys.API_KEY);
+    return (await fetchPassword(PasswordKeys.API_KEY)) || '';
```

Comment:
Is it necessary to return an empty string? I'm always of the mind that, if you're `get`-ting or `fetch`-ing something, if there is no value, the representation for no value is `null`.

Reply:
That's valid, I'll change it in the next PR; basically the UI is ready once the apiKey value is not undefined anymore, but I want to add a pre-loading step on the configuration screens to avoid this and also show a loading instead of empty fields